lib/src/kbasesearchengine/events/handler/WorkspaceEventHandler.java

```diff
@@ -644,4 +670,153 @@ private static String toWSRefPath(final List<GUID> objectRefPath) {
+                        getObjectInfo(Long.decode(accessGrpId), Long.decode(objectId));
+                String latestName = objInfo.getInfos().get(0).getE2();
+
+                if( ev.getNewName().equals(latestName) )
+                    return ev;
+                else {
+                    return StatusEvent.
+                            getBuilder(ev.getStorageObjectType().get(), ev.getTimestamp(), eventType).
+                            withNullableAccessGroupID(Integer.decode(accessGrpId)).
+                            withNullableObjectID(objectId).
+                            withNullableVersion(ev.getVersion().get()).
+                            withNullableNewName(latestName).build();
+                }
+            } catch (IOException ex) {
+                throw handleException(ex);
+            } catch (JsonClientException ex) {
+                throw handleException(ex);
+            }
+        }
+        // update a (UN)PUBLISH_ALL_VERSIONS event
+        if (eventType.equals(StatusEventType.PUBLISH_ALL_VERSIONS) ||
+                eventType.equals(StatusEventType.UNPUBLISH_ALL_VERSIONS )) {
+            try {
+                // get globalread permission value
+                final String isPublic = getWorkspaceInfo(Long.decode(accessGrpId)).getE7();
+
+                // n = no permissions = public access
+                Boolean latestIsPublic = (isPublic == "n") ? true: false;
+
+                if (latestIsPublic) {
+                    return StatusEvent.
+                            getBuilder(ev.getStorageObjectType().get(), ev.getTimestamp(),
+                                    StatusEventType.PUBLISH_ALL_VERSIONS).
+                            withNullableAccessGroupID(Integer.decode(accessGrpId)).
+                            withNullableObjectID(objectId).
+                            withNullableVersion(ev.getVersion().get()).
+                            withNullableisPublic(latestIsPublic).build();
+                } else {
+                    return StatusEvent.
+                            getBuilder(ev.getStorageObjectType().get(), ev.getTimestamp(),
+                                    StatusEventType.UNPUBLISH_ALL_VERSIONS).
+                            withNullableAccessGroupID(Integer.decode(accessGrpId)).
+                            withNullableObjectID(objectId).
+                            withNullableVersion(ev.getVersion().get()).
+                            withNullableisPublic(latestIsPublic).build();
+                }
+
+            } catch (IOException ex) {
+                throw handleException(ex);
+            } catch (JsonClientException ex) {
+                throw handleException(ex);
+            }
+        }
+        // update a (UN)DELETE_ALL_VERSIONS event
+        if (eventType.equals(StatusEventType.DELETE_ALL_VERSIONS) ||
+                eventType.equals(StatusEventType.UNDELETE_ALL_VERSIONS )) {
+            Map<String, Object> command;
```

Comment:
What if the workspace is deleted? the list_objects call will fail

Reply:
Added this check.